validator/keymanager/v2/direct/direct.go

```diff
@@ -73,12 +85,18 @@ func DefaultConfig() *Config {
 
 // NewKeymanager instantiates a new direct keymanager from configuration options.
 func NewKeymanager(ctx context.Context, wallet Wallet, cfg *Config) (*Keymanager, error) {
-	return &Keymanager{
+	k := &Keymanager{
 		wallet:            wallet,
 		cfg:               cfg,
 		mnemonicGenerator: &EnglishMnemonicGenerator{},
 		keysCache:         make(map[[48]byte]bls.SecretKey),
-	}, nil
+	}
+	if wallet.PasswordsDir() != "" {
+		if err := k.initializeSecretKeysCache(); err != nil {
```

Comment:
A comment here will be nice. (eg. what do we have to initialize secret keys cache if pw not provided...)